server/datastore/mysql/software.go

```diff
@@ -332,11 +332,18 @@ func (d *Datastore) AllSoftwareWithoutCPEIterator(ctx context.Context) (fleet.So
 }
 
 func (d *Datastore) AddCPEForSoftware(ctx context.Context, software fleet.Software, cpe string) error {
+	_, err := addCPEForSoftwareDB(ctx, d.writer, software, cpe)
+	return err
+}
+
+func addCPEForSoftwareDB(ctx context.Context, exec sqlx.ExecerContext, software fleet.Software, cpe string) (uint, error) {
```

Comment:
A little refactoring for the test, to get the CPE id as result of the call to the private func only.